src/algorithm/nest_cfgs.rs

```diff
@@ -71,6 +75,92 @@ pub trait CfgView<T> {
     fn successors(&self, node: T) -> Self::Iterator<'_>;
     /// Returns an iterator over the predecessors of the specified basic block.
     fn predecessors(&self, node: T) -> Self::Iterator<'_>;
+
+    /// Given an entry edge and exit edge defining a SESE region, mutates the
+    /// Hugr such that all nodes between these edges are placed in a nested CFG.
+    /// Returns the newly-constructed block (containing a nested CFG).
+    ///
+    /// # Panics
+    /// May panic if the two edges do not constitute a SESE region.
+    fn nest_sese_region(&mut self, entry_edge: (T, T), exit_edge: (T, T)) -> T;
```

Comment:
It is possible we should return some kind of Result, but it's not really clear what. The algorithm here only calls this for single-entry-single-exit regions identified by the analysis, so this should not fail. Perhaps a more complex splitting/merging CfgNodeMap might want to fail in some cases - but *that* I think we should leave for when a fuller splitting/merging CfgNodeMap is developed, i.e. not this PR.